sql/modules/Voucher.sql

```diff
@@ -90,17 +92,23 @@ $$
                 JOIN batch_class bc ON (bc.id = v.batch_class)
                 JOIN account_link l ON (a.chart_id = l.account_id)
                 JOIN ap ON (ap.id = a.trans_id)
-                JOIN entity_credit_account cr
-                        ON (ap.entity_credit_account = cr.id)
-                JOIN company co ON (cr.entity_id = co.entity_id)
+                JOIN entity_credit_account eca
+                        ON (ap.entity_credit_account = eca.id)
+                -- If join only with Company, we can't show correct data for person
+                -- the best way is join with entity instead of joining both person and company
+                JOIN entity e ON (eca.entity_id = e.id)
                 WHERE v.batch_id = in_batch_id
                         AND a.voucher_id = v.id
                         AND (bc.class like 'payment%' AND l.description = 'AP')
-                GROUP BY v.id, a.source, cr.meta_number, co.legal_name ,
+                GROUP BY v.id, ap.invoice, a.source, eca.meta_number, e.name,
                         v.batch_id, v.trans_id, a.transdate, bc.class
 
                 UNION ALL
-                SELECT v.id, false, a.source, a.memo,
+                -- Memo will always empty for batch receipt/payment
```

Comment:
Hmmm. Same remark: after the change, there's no `memo` field reference except for this comment. I think this should be in the commit message.